### DISCUSSION TITLE
How to compare XDocument while ignoring the order of elements.

### DISCUSSION THREAD

**Discussion by ghost** [fluentassertions/fluentassertions#1925] [Q&A] [answered]
Hi ✌️ 

I'm trying to compare two `XDocument` using the `BeEquivalentTo`.
```cs
using System.Xml.Linq;
using FluentAssertions;
using FluentAssertions.Execution;

const string expectedXml = @"
<root>
    <a></a>
    <b></b>
    <c></c>
</root>
";

const string actualXml = @"
<root>
    <b></b>
    <a></a>
    <c></c>
</root>
";

var actual = XDocument.Parse(actualXml);
var expected = XDocument.Parse(expectedXml);

actual.Should().BeEquivalentTo(expected); // throws
```

I was hopping to have the same behavior as the collection form of `BeEquivalentTo`, where the order does not matter, but the method is using `XNode.DeepEquals`, where the order matters.
Note that there might be an issue with .net itself, because according to [this post](https://github.com/dotnet/dotnet-api-docs/issues/830), the order in `XNode.DeepEquals` should not matter.

My question is : is there an elegant way of testing this ? I thought of somehow manually ordering all the elements, but it might take a while for large documents.

**Comment by dennisdoomen** [comment, marked as answer]
`BeEquivalentTo` uses it own implementation instead of `XNode.DeepEquals` to verify equivalency. But since the order of elements in XML _is_ significant, we decided to respect that aspect of XML.

**Reply by ghost** [reply]
Got it. Might need to use something like Collection.Satisfy to replace.
